crates/bevy_ecs/macros/src/lib.rs

```diff
@@ -177,91 +177,105 @@ fn get_idents(fmt_string: fn(usize) -> String, count: usize) -> Vec<Ident> {
 }
 
 #[proc_macro]
-pub fn impl_query_set(_input: TokenStream) -> TokenStream {
+pub fn impl_param_set(_input: TokenStream) -> TokenStream {
     let mut tokens = TokenStream::new();
-    let max_queries = 4;
-    let queries = get_idents(|i| format!("Q{}", i), max_queries);
-    let filters = get_idents(|i| format!("F{}", i), max_queries);
-    let mut query_fn_muts = Vec::new();
-    for i in 0..max_queries {
-        let query = &queries[i];
-        let filter = &filters[i];
-        let fn_name = Ident::new(&format!("q{}", i), Span::call_site());
+    let max_params = 4;
```

Comment:
I could see this being too low, now that there is also reason to also put resources in the set. How about bumping this to 8 or more?

Reply:
I agree; I think 8 is a more reasonable number here, and users will naturally try to avoid abuse of this where possible.

Reply:
I also think it might be interesting to add some flag to the SystemParam derive that would make a custom struct into a ParamSet with each of its fields being accessible by get_*fieldname* methods. Will be out of scope for this pr but should be pretty simple to implement for anyone with some (better than mine) proc macro skills.